src/Avalonia.Controls/Grid/Grid.cs

```diff
@@ -0,0 +1,2373 @@
+        {
+            for (int i = 0; i < grid._definitionsU.Length; i++)
+                if (grid._definitionsU[i].SharedSizeGroup != null)
+                    grid._definitionsU[i].UpdateSharedScope();
+            for (int i = 0; i < grid._definitionsV.Length; i++)
+                if (grid._definitionsV[i].SharedSizeGroup != null)
+                    grid._definitionsV[i].UpdateSharedScope();
+        }
+
+        /// <summary>
+        /// Defines the Column attached property.
+        /// </summary>
+        public static readonly AttachedProperty<int> ColumnProperty =
+            AvaloniaProperty.RegisterAttached<Grid, Control, int>(
+                "Column",
+                validate: ValidateColumn);
+
+        /// <summary>
+        /// Defines the ColumnSpan attached property.
+        /// </summary>
+        public static readonly AttachedProperty<int> ColumnSpanProperty =
+            AvaloniaProperty.RegisterAttached<Grid, Control, int>("ColumnSpan", 1);
+
+        /// <summary>
+        /// Defines the Row attached property.
+        /// </summary>
+        public static readonly AttachedProperty<int> RowProperty =
+            AvaloniaProperty.RegisterAttached<Grid, Control, int>(
+                "Row",
+                validate: ValidateRow);
+
+        /// <summary>
+        /// Defines the RowSpan attached property.
+        /// </summary>
+        public static readonly AttachedProperty<int> RowSpanProperty =
+            AvaloniaProperty.RegisterAttached<Grid, Control, int>("RowSpan", 1);
+
+        public static readonly AttachedProperty<bool> IsSharedSizeScopeProperty =
+            AvaloniaProperty.RegisterAttached<Grid, Control, bool>("IsSharedSizeScope", false);
+
+        internal static readonly AttachedProperty<SharedSizeScope> PrivateSharedSizeScopeProperty =
+            AvaloniaProperty.RegisterAttached<Grid, Control, SharedSizeScope>("&&PrivateSharedSizeScope", null, inherits: true);
+
+        /// <summary>
+        /// Defines the <see cref="ShowGridLines"/> property.
+        /// </summary>
+        public static readonly StyledProperty<bool> ShowGridLinesProperty =
+            AvaloniaProperty.Register<Grid, bool>(
+                nameof(ShowGridLines),
+                defaultValue: false);
+
+        /// <summary>
+        /// ShowGridLines property.
+        /// </summary>
+        public bool ShowGridLines
+        {
+            get { return GetValue(ShowGridLinesProperty); }
+            set { SetValue(ShowGridLinesProperty, value); }
+        }
+
+        /// <summary>
+        /// Gets or sets the columns definitions for the grid.
+        /// </summary>
+        public ColumnDefinitions ColumnDefinitions
+        {
+            get
+            {
+                if (_columnDefinitions == null)
+                {
+                    ColumnDefinitions = new ColumnDefinitions();
+                }
+
+                return _columnDefinitions;
+            }
+            set
+            {
```

Comment:
@grokys This and `RowDefinition`'s setter are horrible but i ran out of ideas on how to properly mirror those two on their array counterparts...